### PR TITLE
feat: Add link column to CSV file for org members

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4738
Screenshot 2024-11-17 at 02 13 58

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: packages/features/users/components/UserTable/UserListTable.tsx [hunk 1]
```diff
@@ -475,7 +481,7 @@ export function UserListTable() {
                 className="sm:max-w-64 max-w-full"
               />
             </div>
-            <div className="flex flex-wrap items-center gap-2">
+            <div className="flex flex-wrap items-center justify-between gap-2">
```
Comment (hbjORbj) [inline packages/features/users/components/UserTable/UserListTable.tsx:484]: improve styling:

Before
Screenshot 2024-11-17 at 01 10 36

After
after

File: packages/lib/csvUtils.ts [hunk 2]
```diff
@@ -86,20 +82,38 @@ export const generateHeaderFromReactTable = (
...
+    throw new Error("The header is empty.");
+  }
+
+  const REQUIRED_HEADERS = ["Members", "Link", "Role", "Teams"] as const;
+  // Validate required headers are present and in correct order
+  const firstFourHeaders = headers.slice(0, REQUIRED_HEADERS.length);
+  if (!REQUIRED_HEADERS.every((header, index) => header === firstFourHeaders[index])) {
+    throw new Error(
+      `Invalid headers structure. Expected headers to start with: ${JSON.stringify(REQUIRED_HEADERS)}`
+    );
```
Comment (hbjORbj) [inline packages/lib/csvUtils.ts:111]: header validation

File: packages/lib/csvUtils.ts [hunk 3]
```diff
@@ -86,20 +82,38 @@ export const generateHeaderFromReactTable = (
...
+  const memberIndex = headerNames.findIndex((name) => name === MEMBERS_COLUMN);
+  if (memberIndex > -1) {
+    headerNames.splice(memberIndex + 1, 0, LINK_COLUMN);
+  }
+
   return headerNames;
 };
 
-export const generateCsvRaw = (
+export const generateCsvRawForMembersTable = (
```
Comment (hbjORbj) [inline packages/lib/csvUtils.ts:96]: renaming

File: packages/lib/csvUtils.ts [hunk 4]
```diff
@@ -86,20 +82,38 @@ export const generateHeaderFromReactTable = (
...
   });
 
+  // Add "Link" column (member's public page)
+  const MEMBERS_COLUMN = "Members";
+  const LINK_COLUMN = "Link";
+  const memberIndex = headerNames.findIndex((name) => name === MEMBERS_COLUMN);
+  if (memberIndex > -1) {
+    headerNames.splice(memberIndex + 1, 0, LINK_COLUMN);
+  }
+
```
Comment (hbjORbj) [inline packages/lib/csvUtils.ts:92]: adding link column

**Review by PeerRich — APPROVED** [review]

**Pull request merged**